src/v/debug_bundle/types.h

```diff
@@ -88,6 +88,18 @@ struct partition_selection {
 std::ostream& operator<<(std::ostream& o, const partition_selection& p);
 
 using k8s_namespace = named_type<ss::sstring, struct k8s_namespace_tag>;
+using k8s_label_key = named_type<ss::sstring, struct k8s_label_key_tag>;
+using k8s_label_value = named_type<ss::sstring, struct k8s_label_value_tag>;
```

Comment:
why do we need anything `k8s` specific in the tree? can't we provide a mechanism to pass arbitrary key-value pairs?

Reply:
but why does redpanda need to validate that?

Reply:
I'm validating length and character types.  I'm not doing a full validation and allowing rpk to do it.  The length/character type check is present to ensure no 'weird' characters are present and the length check is to ensure we aren't passing a massive string with null characters in it.

This does remind me to add in some additional checks for the other strings we handle (password/username etc).

Reply:
> The length/character type check is present to ensure no 'weird' characters are present and the length check is to ensure we aren't passing a massive string with null characters in it.

sounds like these are generic checks, not specific to k8s? i'm all for validation, but k8s specific?

Reply:
Actually in the process of removing them.  Gave it a thought and remembered the validate_no_control function.  Going to implement using that against all provided strings and remove these checks.